src/components/TodoList/TodoList.tsx

```diff
@@ -1,13 +1,30 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import cn from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodosActions } from '../../features/currentTodo';
+import { getVisibleTodos } from '../../utils/getVisibleTodos';
 
 export const TodoList: React.FC = () => {
-  return (
-    <>
+  const { todos, currentTodo } = useAppSelector(state => state);
```

Comment:
Don't access parts of the state like this. It will cause your component to re-render on every state change.